compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -573,6 +573,9 @@ public BSymbol lookupLangLibMethod(BType type, Name name) {
             case TypeTags.XML_PI:
             case TypeTags.XML_TEXT:
                 bSymbol = lookupLangLibMethodInModule(symTable.langXmlModuleSymbol, name);
+                if (bSymbol == symTable.notFoundSymbol && type.tag == TypeTags.XML_TEXT) {
```

Comment:
Shall we add this new lookup logic only to XML_TEXT case
I think it will make it more evident that we are only intending to lookup string methods for XML_TEXT Tag.

What I'm suggesting is to have `bSymbol = lookupLangLibMethodInModule(symTable.langXmlModuleSymbol, name); break;` after L:573

Reply:
Done